plugin/inc/views/view.class.php

```diff
@@ -1445,14 +1445,41 @@ protected function evalIfFunction($string, $global_variables = []): string
         return $string;
     }
 
-    public static function modifyTitleBasedOnUnpaidInvoices(array $rowData, string &$title, string &$color): void
+    public static function getPartnerTitleBasedOnUnpaidInvoices(int $numberOfUnpaidInvoices, $valueOfUnpaidInvocies, bool $withLineBreaks = false): string
     {
-        if ($rowData['numar_facturi_neplatite'] >= 2) {
-            $color  = "red";
-            $title .= "\r\n\r\nPartenerul are " . $rowData['numar_facturi_neplatite'] . " facturi neplătite " . sprintf(_t('in value of %s RON'), number_format($rowData['unpaid_invoices_value'], 2));
-        } elseif ($rowData['numar_facturi_neplatite'] == 1) {
-            $color  = "orange";
-            $title .= "\r\n\r\nPartenerul are o factură neplătită " . sprintf(_t('in value of %s RON'), number_format($rowData['unpaid_invoices_value'], 2));
+
+        if ($numberOfUnpaidInvoices < 1) {
+            return '';
+        }
+
+        $title     = '';
+        $arguments = $numberOfUnpaidInvoices === 1 ? [$valueOfUnpaidInvocies] : [$numberOfUnpaidInvoices, $valueOfUnpaidInvocies];
+        if ($withLineBreaks) {
+            $title .= "\r\n\r\n";
         }
+
+        $title .= sprintf(
+            _tn('Client has an unpaid invoice in value of %s RON', 'Client has %s unpaid invoices in value of %s RON', $numberOfUnpaidInvoices),
+            ...$arguments
+        );
+
+        return $title;
     }
+
+    public static function getPartnerStyleBasedOnUnpaidInvoices(int $numberOfUnpaidInvoices, $style = ''): string
+    {
+        $style = 'color: green;';
+        if ($numberOfUnpaidInvoices === 1) {
```

Comment:
This would have more sense:
```suggestion
    public static function getPartnerStyleBasedOnUnpaidInvoices(int $numberOfUnpaidInvoices, $style = 'color: green;'): string
    {
        if ($numberOfUnpaidInvoices === 1) {
```